### PR TITLE
chore: go-kit bump 0.20.2 -> 0.23.2

### PR DESCRIPTION
**Pull request by Sidddddarth**
# Description

Bump go-kit version from 0.20.2 to 0.23.2.

## Linear Ticket

[Fixes PIPE-885](https://linear.app/rudderstack/issue/PIPE-885/fix-rudder-go-kit-update-brakes-rudder-server)

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

**Review by cisse21 — APPROVED** [review]

File: jobsdb/jobsdb_test.go [hunk 1]
```diff
@@ -36,251 +33,196 @@ import (
...
 					jd.Stop()
-					Expect(jd.backgroundGroup.Wait()).To(BeNil())
+					require.NoError(t, jd.backgroundGroup.Wait())
 					wg.Done()
 				}()
 			}
 			wg.Wait()
-		})
-
-		It("can call Start & Stop in parallel without problems", func() {
```
Comment (achettyiitr) [inline jobsdb/jobsdb_test.go:267]: Don't we need this test?

**Review by achettyiitr — APPROVED** [review]

File: jobsdb/jobsdb_test.go [hunk 2]
```diff
@@ -36,251 +33,196 @@ import (
 	. "github.com/rudderlabs/rudder-server/utils/tx" //nolint:staticcheck
 )
 
-var _ = Describe("Calculate newDSIdx for internal migrations", Ordered, func() {
-	BeforeAll(func() {
+func TestComputeInsertIdx(t *testing.T) {
```
Comment (fracasula) [inline jobsdb/jobsdb_test.go:36]: God bless you Siddarth :joy: (I don't like Ginkgo :sweat:)

Comment (Sidddddarth) [reply jobsdb/jobsdb_test.go:36]: same😅

File: jobsdb/jobsdb_test.go (hunk 1 above)
Comment (Sidddddarth) [reply jobsdb/jobsdb_test.go:267]: we do yeah.
missed it while cleaning up, my bad.

Comment (Sidddddarth) [reply jobsdb/jobsdb_test.go:267]: added it back

**Review by fracasula — APPROVED** [review]

**Pull request merged**